Documentation for existing cluster importing (#131)
kubernetes-sigs/kube-scheduler-simulator: README.md

[Scheduler Configuration | Kubernetes](https://kubernetes.io/docs/reference/scheduling/config/)

> You can pass a KubeSchedulerConfiguration file via the environment variable `KUBE_SCHEDULER_CONFIG_PATH` and the simulator will start kube-schedule  with that configuration.

Reviewer: ```suggestion
You can pass a path to KubeSchedulerConfiguration via the environment variable `KUBE_SCHEDULER_CONFIG_PATH` and the simulator will start kube-scheduler with that configuration.
```

Author: Oops... This line is probably an unrelated line engulfed by bad rebase. (But I fixed)

Note: changes to any fields other than `.profiles` are disabled on simulator, since they do not affect the results of the scheduling.

![configure scheduler](simulator/docs/images/schedulerconfiguration.png)

If you want to use your custom plugins as out-of-tree plugins in the simulator, please follow [this doc](simulator/docs/how-to-use-custom-plugins/README.md).

## Getting started

> Read more about environment variables being used in simulator server
> [here.](./simulator/docs/env-variables.md)

Reviewer: ```suggestion
You can find more information about environment variables available in the simulator server
[here.](./simulator/docs/env-variables.md)

```

To run the frontend, please see [README.md](./web/README.md) on ./web dir.

## [Beta] Existing cluster importing

The simulator can import existing clusters.
This allows for batch inclusion of resources from external clusters.

> It is enabled by an `EXTERNAL_IMPORT_ENABLED` environment variables is `1`.

Reviewer: ```suggestion
It is enabled by an `EXTERNAL_IMPORT_ENABLED` environment variables is `1`.

Currently, the simulator tries to import ALL resources related to scheduling from your cluster,
so, please be careful (or avoid) using it in clusters with large amounts of resources. 
```

> You can import resources on your cluster to the simulator by using the following environment variables `KUBE_SCHEDULER_CONFIG_PATH` and `KUBECONFIG`.

Reviewer: ```suggestion
You can import resources on your cluster to the simulator by using the environment variable `KUBECONFIG`.
(And if you would like to use your cluster's scheduler configuration, you may want to use `KUBE_SCHEDULER_CONFIG_PATH`.)
```

To run the frontend, please see [README.md](./web/README.md) on ./web dir.

> ## Existing cluster importing

Reviewer: Let's specify this feature as Beta for now. I think we need to improve this more to mark this feature as Stable. #142

```suggestion
## [Beta] Existing cluster importing
```

> The simulator can import existing clusters.
> This allows for batch inclusion of resources from external clusters.
> It is enabled by an `EXTERNAL_IMPORT_ENABLED` environment variables is `1`.
> 
> The following environment variables configure this function:

Reviewer: ```suggestion

You can import resources on your cluster to the simulator by using the following environment variables.
```

> - `EXTERNAL_IMPORT_ENABLED`:If it is `1`, this importing function is enabled.
>   The scheduler starts importing from external clusters as soon as it starts.

Reviewer: ```suggestion
- `EXTERNAL_IMPORT_ENABLED`: If you set `1` to `EXTERNAL_IMPORT_ENABLED`, resources on your cluster will be imported right after the simulator starts.
```

- `KUBECONFIG`:The cluster from which you are importing is indicated by this environment variable.
  [Read more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)

> - `KUBE_SCHEDULER_CONFIG_PATH`:This is an optional. The simulator can't import the scheduler configuration from existing cluster via `kube-apiserver`.

Reviewer: I think we don't need to write a doc for `KUBE_SCHEDULER_CONFIG_PATH` here.
We should write it on #138.
cc @pravarag

Reviewer 2: Also, @sanposhiho how about we include `KUBECONFIG` and `EXTERNAL_IMPORT_ENABLED` fields as well in #138 ? We can then refer it all together in the README.md file.

Reviewer: 👍 

Having the same explanation in two places can lead to forgetting it when updating the doc in the future. So, I think it's better to add doc about KUBECONFIG and EXTERNAL_IMPORT_ENABLED in #138 and just add the link to docs/env-variables.md from here as @pravarag suggested.

maybe like:

```markdown

## [Beta] Existing cluster importing

You can import resources on your cluster to the simulator by using the following environment variables `EXTERNAL_IMPORT_ENABLED` and `KUBECONFIG`.
For more information about those variables, please refer [docs/env-variables.md](docs/env-variables.md).
```

Author: yup, thanks! I have replaced all of your suggestions. So, I wait for #138 to be finished and merged.

/hold again

To run the frontend, please see [README.md](web/README.md) on ./web dir.

## [Beta] Existing cluster importing

> **PLEASE AVOID** using it in clusters with large amounts of resources.
> The simulator will try to import ALL resources related to scheduling from your cluster and it may badly affect your cluster's api-server.

Reviewer: Sorry, I know it's suggested by the past me, but now I know it isn't needed to be mentioned.  

```suggestion
```

To run the frontend, please see [README.md](web/README.md) on ./web dir.

> ## [Beta] Existing cluster importing

Author: As an aside, we should have said "external cluster", I think.
But I named that service as `replicateexistingcluster`...

Reviewer: > we should have said "external cluster"

Tashikani, it seems better 😓 
You can rename it before releasing v0.1.0 if you like. But, it is up to you; not big problem here.

> The simulator can import existing clusters.
> This allows for batch inclusion of resources from external clusters.

Reviewer: ```suggestion
The simulator can import resources from your cluster.
```

> It is enabled by an `EXTERNAL_IMPORT_ENABLED` environment variables is `1`.

Reviewer: ```suggestion
You can use it by setting an `EXTERNAL_IMPORT_ENABLED` environment variable to `1`.
```

> Please be careful (or avoid) using it in clusters with large amounts of resources.
> The simulator tries to import ALL resources related to scheduling from your cluster. 

Reviewer: ```suggestion
PLEASE AVOID using it in clusters with large amounts of resources.
The simulator will try to import ALL resources related to scheduling from your cluster and it may badly affect your cluster's api-server.
```

Reviewer: Could you move this to around L116. It's so important one.

> You can import resources on your cluster to the simulator by using the environment variable `KUBECONFIG`.

Reviewer: ```suggestion
You need to have the kubeconfig to import resources on your cluster.
The simulator tries to read the kubeconfig file on the environment variable `KUBECONFIG`.
```

> (And if you would like to use your cluster's scheduler configuration, you may want to use `KUBE_SCHEDULER_CONFIG_PATH`.)

Reviewer: Seems not needed.

```suggestion
```

## Existing cluster importing
The simulator can import existing clusters.
This allows for batch inclusion of resources from external clusters.
It is enabled by an `EXTERNAL_IMPORT_ENABLED` environment variables is `1`.

The following environment variables configure this function:
- `EXTERNAL_IMPORT_ENABLED`:If it is `1`, this importing function is enabled.
  The scheduler starts importing from external clusters as soon as it starts.
  - `KUBECONFIG`:The cluster from which you are importing is indicated by this environment variable.
    [Read more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
    - `KUBE_SCHEDULER_CONFIG_PATH`:This is an optional. The simulator can't import the scheduler configuration from existing cluster via `kube-apiserver`.
      If you set the file path to this variable, then you could import the scheduler configuration.

## Contributing

see [CONTRIBUTING.md](CONTRIBUTING.md)
TULATOR_EXTERNAL_IP=your.server.ip` before running `docker-compose up -d`.

Note: Insufficient memory allocation may cause problems in building the image.

> Please allocate enough memory in that case.

Reviewer: Seems a bad rebase around here as well.
